File: calculaPrimo.dart
calculaPrimo(numero) {
    bool primo = false;
    double metade = (numero / 2 + 1);
    int metadeDivisao = metade.round();
    for(int x = 2; x < metadeDivisao; x++) {
        if(numero % x == 0) {
            primo = false;
            break;
        } else {
            primo = true;
        }
    }
    return primo;
}

main() {
    List listaPrimos = [];
    for(int x = 1; x < 1000000; x++) {
        print("Checando $x");
        if(calculaPrimo(x)) {
            listaPrimos.add(x);
        }
    }
    print("Confira a lista");
    print(listaPrimos);
    print("São ${listaPrimos.length} números primos");
}

File: calculaPrimo.py
from math import ceil

def calculaPrimo(numero):
    primo = False
    metade = (numero / 2 + 1)
    metadeDivisao = ceil(metade)
    for x in range(2, metadeDivisao):
        if (numero % x == 0):
            primo = False
            break
        else:
            primo = True
    return primo

listaPrimos = []
for y in range(1, 100000):
    print(f"Checando {y}...")
    if calculaPrimo(y):
        print(f"{y} é primo")
        listaPrimos.append(y)
print("Confira a lista:")
print(listaPrimos)
print(f"Total de números primos {len(listaPrimos)}")